### ISSUE TITLE
Encoder doesn't get gradients

### ISSUE THREAD

**Issue by oleksost** [lebrice/Sequoia#71]
Taking for example: 

```
likelihood = torch.log_softmax(self._model(observation).logits, dim=1)[0][0]
grad = torch.autograd.grad(
            likelihood, self._model.parameters(),
            retain_graph=True, allow_unused=True,
        )
for (name, p ),g in zip(self._model.named_parameters(), grad): print(name, g)
```
for the encoder parameters, the gradient is None.

This can be reproduced on the 'reg_method' branch with these parameters for example: 

`"--setting", "task_incremental_rl", "--method", "baseline", "--ewc.coef", "1", "--verbose", "1", "--max_epochs", "1", "--batch_size", "32", "--max_steps", "1000", "--dataset", "cartpole", "--wandb", "test", "--no_wandb", "--FIM_representation", "block_diagonal", "--ewc_critic", "1", "--rl_algo", "DQN", "--ewc_coefficient", "100", "--nb_tasks", "5", "--train_task_schedule_path", "/Users/oleksostapenko/Projects/SSCL/sequoia/schedules/train_schedule.json"]`.

Train schedule is something like this: 
```
{              
    "0":   {"gravity": 5, "length": 0.8},
    "200":   {"gravity": 20, "length": 0.2}
}

```
